src/boost_histogram/_internal/hist.py

```diff
@@ -376,24 +385,38 @@ def _reduce(self, *args):
         return self._new_hist(self._hist.reduce(*args))
 
     def __copy__(self):
-        other = self._new_hist(copy.copy(self._hist))
-        return other
+        return self._new_hist(copy.copy(self._hist))
 
     def __deepcopy__(self, memo):
-        other = self.__class__.__new__(self.__class__)
-        other._hist = copy.deepcopy(self._hist, memo)
-        other.metadata = copy.deepcopy(self.metadata, memo)
-        other.axes = other._generate_axes_()
-        return other
+        return self._new_hist(copy.deepcopy(self._hist), memo=memo)
 
     def __getstate__(self):
-        state = {"_hist": self._hist, "metadata": self.metadata}
+        """
+        Version 0.8: metadata added
+        Version 0.11: version added and set to 0. metadata/_hist replaced with dict.
+
+        ``dict`` contains __dict__ without "axes" and "_hist"
+        """
+        local_dict = copy.copy(self.__dict__)
+        del local_dict["axes"]
+        state = {"dict": local_dict, "version": 0}
         return state
 
     def __setstate__(self, state):
-        self._hist = state["_hist"]
-        self.metadata = state.get("metadata", None)
-        self.axes = self._generate_axes_()
+        if "version" not in state:
```

Comment:
If state is a tuple, you can place the new version number at the front of the tuple. To discriminate between old and new, you can check whether `state` is a dict or a tuple.